File: problems/java/arrays_hashing/valid_anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        System.out.println((int)'a');
        int[] hashS = new int [26];
        int[] hashT = new int [26];
        int lenS = s.length();
        int lenT = t.length();
        
        if(lenS != lenT){
            return false;
        }
        
        for(int i=0; i<lenS; i++){
            int curS = s.charAt(i)-97;
            int curT = t.charAt(i)-97;

            hashS[curS] += 1;
            hashT[curT] += 1;
        }

        for(int i=0; i<26; i++){
            if(hashS[i] != hashT[i]){
                return false;
            }
        }

        return true;
    }
}

File: python/arrays_hashing/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hm = {}
        for c in s:
            if c in hm:
                hm[c] += 1
            else:
                hm[c] = 1
        
        for c in t:
            if c in hm:
                hm[c] -= 1
            else:
                return False
        
        for (key, count) in hm.items():
            if count != 0:
                return False
        return True